src/common/SmartUrlInput.js

```diff
@@ -85,9 +85,10 @@ export default class SmartUrlInput extends PureComponent<Props, State> {
       onSubmitEditing,
       enablesReturnKeyAutomatically,
     } = this.props;
-    const { value } = this.state;
+    let { value } = this.state;
     const useFullAppend = value.indexOf('.') === -1;
-    const showAnyAppend = !value.match(/.+\..+\.+./g); // at least two dots
+    const showAnyAppend = !(defaultValue && value.length === 0) && !value.match(/.+\..+\.+./g); // at least two dots
+    value = defaultValue && value.length === 0 ? defaultValue : value;
```

Comment:
This logic is kind of confusing to read -- partly because it has this slightly complex condition `defaultValue && value.length === 0`, and uses it twice.

Can you do this with something like
```
if (defaultValue && value.length === 0) {
  value = defaultValue;
  showAnyAppend = false;
}
```

Reply:
yup! done 👍